frontend/src/components/camp/picasso/Picasso.vue

```diff
@@ -0,0 +1,414 @@
+          </template>
+        </dialog-activity-edit>
+
+        <h4 class="v-event-title">
+          {{ getActivityName(event) }}
+        </h4>
+
+        <!-- extendBottom: used for resizing -->
+        <div
+          v-if="editable && timed"
+          class="v-event-drag-bottom"
+          @mousedown.stop="extendBottom(event)" />
+      </template>
+    </v-calendar>
+
+    <v-snackbar v-model="isSaving" light>
+      <template v-if="patchError">
+        <v-icon>mdi-alert</v-icon>
+        {{ patchError }}
+      </template>
+      <template v-else>
+        <v-icon class="mdi-spin">mdi-loading</v-icon>
+        {{ $tc('global.button.saving') }}
+      </template>
+    </v-snackbar>
+  </div>
+</template>
+<script>
+import { toRefs } from '@vue/composition-api'
+import useDragAndDrop from './useDragAndDrop.js'
+import { isCssColor } from 'vuetify/lib/util/colorUtils'
+
+import DialogActivityEdit from '@/components/dialog/DialogActivityEdit.vue'
+
+export default {
+  name: 'Picasso',
+  components: {
+    DialogActivityEdit
+  },
+  props: {
+    // period for which to schow picasso
+    period: {
+      type: Object,
+      required: true
+    },
+
+    // list of scheduleEntries
+    scheduleEntries: {
+      type: Array,
+      required: true
+    },
+
+    // false disables drag & drop and disabled edit dialogs
+    editable: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+
+    // v-calendar start: starting date (first day)
+    start: {
+      type: Number,
+      required: true
+    },
+
+    // v-calender end: end date (last day)
+    end: {
+      type: Number,
+      required: true
+    },
+
+    // v-calendar type
+    type: {
+      type: String,
+      required: false,
+      default: 'custom-daily'
+    },
+
+    // v-calendar intervalHeight
+    intervalHeight: {
+      type: Number,
+      required: false,
+      default: 0
+    }
+
+  },
+
+  // emitted events
+  emits: [
+    'changePlaceholder', // triggered continuously while a new entry is being dragged (parameters: startTime, endTime)
+
+    'newEntry', // triggered once when a new entry was created via drag & drop (parameters: startTime, endTime)
+
+    'openEntry' // triggered when an existing entry is clicked on (parameters: entry, newTab)
+  ],
+
+  // composition API setup
+  setup (props, { emit }) {
+    const { editable } = toRefs(props)
+
+    const {
+      entryMouseDown,
+      timeMouseDown,
+      timeMouseMove,
+      timeMouseUp,
+      nativeMouseUp,
+      extendBottom,
+      isSaving,
+      patchError
+    } = useDragAndDrop(editable, emit)
+
+    return {
+      // methods
+      entryMouseDown,
+      timeMouseDown,
+      timeMouseMove,
+      timeMouseUp,
+      nativeMouseUp,
+      extendBottom,
+
+      // data
+      isSaving,
+      patchError
+    }
+  },
+  data () {
+    return {
+      maxDays: 100,
+      entryWidth: 80,
+      value: '',
+      activitiesLoading: true
+    }
+  },
```

Comment:
Ah, ist es legal, `setup()` zu definieren und trotzdem noch die alte Struktur mit `data`, `computed` etc. zu behalten? Ich dachte man müsste dann auch alle `data` etc. in `setup` definieren.

Reply:
Yep, kann man mixen.